### DISCUSSION TITLE
Suggest autocomplete while typing

### DISCUSSION THREAD

**Discussion by gonzalezgarcia** [jspsych/jsPsych#2945] [Q&A]
Hi,

I'm programming an experiment where we ask people to type the name of an object. We would like to display autocomplete options (from a predefined list of words) as participants type. Something similar to this: https://www.w3schools.com/howto/howto_js_autocomplete.asp

I've taken a look at the available plugins but can't seem to find the right option. I was wondering if there's a way to implement such feature within jsPsych.

Best,
Carlos

**Comment by jodeleeuw** [comment]
Hi @gonzalezgarcia,

You could incorporate the code in that example in a custom jsPsych plugin. 

We have [a guide about creating custom plugins](https://www.jspsych.org/7.3/developers/plugin-development/) and I also recorded [a video tutorial](https://www.youtube.com/watch?v=XQcsFwAmbiw) a few years ago. The video tutorial show how to do this in version 6, but the concepts are the same.

For this plugin you could probably start with the survey-text plugin and modify that to meet your needs.

This sounds like a useful plugin to have, so if you do create it please consider adding it to [jspsych-contrib](https://github.com/jspsych/jspsych-contrib).
